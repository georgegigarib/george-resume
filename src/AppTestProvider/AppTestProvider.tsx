import React from 'react'
import { I18nextProvider } from 'react-i18next'
import i18n from '@/locales/i18n.config'
import store from '@/store'
import { Provider } from 'react-redux'
import { BrowserRouter as Router, Routes, Route } from 'react-router-dom'
import { vi } from 'vitest'

vi.mock('swiper/react', () => ({
  Swiper: ({
    children,
    onSlideChange,
  }: {
    children: React.ReactNode
    onSlideChange?: (params: { realIndex: number }) => void
  }) => {
    onSlideChange?.({ realIndex: 0 })
    return <div data-testid="swiper">{children}</div>
  },
  SwiperSlide: ({ children }: { children: React.ReactNode }) => <div data-testid="swiper-slide">{children}</div>,
}))

const AppTestProvider = ({ children }: { children: React.ReactNode }) => {
  return (
    <I18nextProvider i18n={i18n}>
      <Provider store={store}>
        <Router>
          <Routes>
            <Route path="/" element={children} />
          </Routes>
        </Router>
      </Provider>
    </I18nextProvider>
  )
}

export default AppTestProvider
