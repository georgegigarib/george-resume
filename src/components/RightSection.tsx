import { useEffect, useState } from 'react';
import Box from "./Box";

const RightSection = () => {
  const boxes = [
    { startRow: 13, spanRow: 20, startColumn: 1, spanColumn: 13, content: "Box 12" },
    { startRow: 21, spanRow: 12, startColumn: 14, spanColumn: 36, content: "Box 13" },
    { startRow: 33, spanRow: 35, startColumn: 22, spanColumn: 22, content: "Box 14" },
    { startRow: 51, spanRow: 8, startColumn: 14, spanColumn: 8, content: "Box 15" },
    { startRow: 33, spanRow: 10, startColumn: 44, spanColumn: 30, content: "Box 16" },
    { startRow: 43, spanRow: 18, startColumn: 44, spanColumn: 18, content: "Box 17" },
    { startRow: 43, spanRow: 40, startColumn: 62, spanColumn: 12, content: "Box 18" },
  ];

  const smBoxes = [
    { startRow: 11, spanRow: 20, startColumn: 3, spanColumn: 15, content: "Small Box 16" },
    { startRow: 8, spanRow: 3, startColumn: 7, spanColumn: 6, content: "Box 17" },
    { startRow: 9, spanRow: 2, startColumn: 13, spanColumn: 5, content: "18" },
    { startRow: 9, spanRow: 15, startColumn: 18, spanColumn: 3, content: "Small Box 14" },
    { startRow: 5, spanRow: 4, startColumn: 13, spanColumn: 8, content: "Small Box 15" },
  ];

  const [activeBoxes, setActiveBoxes] = useState(boxes);

  const updateBoxes = () => {
    const width = window.innerWidth;
    if (width <= 768) {
      setActiveBoxes(smBoxes);
    } else {
      setActiveBoxes(boxes);
    }
  };

  useEffect(() => {
    updateBoxes();

    window.addEventListener('resize', updateBoxes);
    return () => {
      window.removeEventListener('resize', updateBoxes);
    };
  }, []);

  return (
    <div className="grid grid-cols-100 grid-rows-100 gap-4">
      {activeBoxes.map((box, index) => (
        <Box 
          key={index}
          startRow={box.startRow}
          startColumn={box.startColumn}
          spanRow={box.spanRow}
          spanColumn={box.spanColumn}
          content={box.content} 
        />
      ))}
    </div>
  );
}

export default RightSection;
