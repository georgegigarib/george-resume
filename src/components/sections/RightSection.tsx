import { useEffect, useState } from 'react'
import Box from '@/components/organisms/Box'
import { useTranslation } from 'react-i18next'
import ExampleComponent from '@/components/organisms/ExampleComponente'
import SpotifyPlaylist from '@/components/organisms/SpotifyPlaylist'
import LatestSong from '@/components/organisms/LatestSong'
import { useIsMobile } from '@/hooks/useIsMobile'
import YoutubeVideo from '@/components/organisms/YoutubeVideo'

const RightSection = () => {
  const { t, i18n } = useTranslation()
  const isMobile = useIsMobile()
  const [activeBoxes, setActiveBoxes] = useState<
    {
      startRow: number
      spanRow: number
      startColumn: number
      spanColumn: number
      content: React.ReactNode
    }[]
  >([])

  useEffect(() => {
    const mobileBoxes = [
      { startRow: 19, spanRow: 21, startColumn: 9, spanColumn: 17, content: <SpotifyPlaylist /> },
      { startRow: 19, spanRow: 13, startColumn: 1, spanColumn: 8, content: <YoutubeVideo /> },
      { startRow: 11, spanRow: 8, startColumn: 11, spanColumn: 8, content: <ExampleComponent /> },
      { startRow: 3, spanRow: 16, startColumn: 19, spanColumn: 7, content: <ExampleComponent /> }
    ]

    const desktopBoxes = [
      { startRow: 13, spanRow: 20, startColumn: 1, spanColumn: 13, content: <ExampleComponent /> },
      { startRow: 21, spanRow: 12, startColumn: 14, spanColumn: 36, content: <ExampleComponent /> },
      { startRow: 33, spanRow: 22, startColumn: 22, spanColumn: 22, content: <SpotifyPlaylist /> },
      { startRow: 48, spanRow: 6, startColumn: 14, spanColumn: 8, content: <ExampleComponent /> },
      { startRow: 33, spanRow: 7, startColumn: 44, spanColumn: 20, content: <LatestSong /> },
      { startRow: 40, spanRow: 10, startColumn: 44, spanColumn: 10, content: <ExampleComponent /> },
      { startRow: 41, spanRow: 14, startColumn: 54, spanColumn: 10, content: <YoutubeVideo /> }
    ]

    setActiveBoxes(isMobile ? mobileBoxes : desktopBoxes)
  }, [isMobile, i18n.language, t])

  return (
    <div className="grid grid-cols-100 grid-rows-100 gap-4">
      {activeBoxes.map((box, index) => (
        <Box
          key={index}
          startRow={box.startRow}
          startColumn={box.startColumn}
          spanRow={box.spanRow}
          spanColumn={box.spanColumn}
          content={box.content}
        />
      ))}
    </div>
  )
}

export default RightSection
