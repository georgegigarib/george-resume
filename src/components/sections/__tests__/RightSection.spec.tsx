import { render } from '@testing-library/react'
import { describe, expect, it, vi } from 'vitest'
import RightSection from '@/components/sections/RightSection'
import AppTestProvider from '@/AppTestProvider/AppTestProvider'

const isMobileMock = vi.fn()

vi.mock('@/hooks/useIsMobile', () => ({
  useIsMobile: () => isMobileMock(),
}))

describe('RightSection component', () => {
  const renderComponent = () =>
    render(
      <AppTestProvider>
        <RightSection />
      </AppTestProvider>
    )

  it('renders correctly on desktop', () => {
    isMobileMock.mockReturnValue(false)

    const { asFragment } = renderComponent()
    expect(asFragment()).toMatchSnapshot()
  })
})
