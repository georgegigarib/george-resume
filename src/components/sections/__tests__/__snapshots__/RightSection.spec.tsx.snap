// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`RightSection component > renders correctly on desktop 1`] = `
<DocumentFragment>
  <div
    class="grid grid-cols-100 grid-rows-100 gap-4"
  >
    <div
      class="rounded-xl bg-transparent  shadow-3d-light dark:shadow-3d-dark transition-all hover:shadow-hover-light dark:hover:shadow-hover-dark "
      style="grid-area: 13 / 1 / span 20 / span 13;"
    >
      <div
        class="flex w-full h-full"
        style="perspective: 1000px;"
      >
        <div
          class="w-full relative transition-all duration-200 ease-linear inter-var"
          style="transform-style: preserve-3d;"
        >
          <div
            class="[transform-style:preserve-3d] [&>*]:[transform-style:preserve-3d] bg-gray-50 group/card dark:hover:shadow-2xl dark:bg-black dark:border-white/[0.2] border-black/[0.1] w-full h-full rounded-xl border"
          >
            <div
              class="text-blue-900 rounded-xl flex justify-center flex-wrap items-center contain-size overflow-hidden h-full border-[2px] bg-app-lightModeBoxes-box11 dark:bg-app-darkModeBoxes-box11 border-app-darkModeBoxes-box11 dark:border-app-lightModeBoxes-box11"
              data-testid="example-container"
            >
              <div
                class="absolute inset-0 flex items-center justify-center"
              >
                <span
                  class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-1yonwsg-MuiCircularProgress-root"
                  role="progressbar"
                  style="width: 40px; height: 40px;"
                >
                  <svg
                    class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
                    viewBox="22 22 44 44"
                  >
                    <circle
                      class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-1a8epmm-MuiCircularProgress-circle"
                      cx="44"
                      cy="44"
                      fill="none"
                      r="20.2"
                      stroke-width="3.6"
                    />
                  </svg>
                </span>
              </div>
              <img
                class="rounded-lg min-h-full min-w-full flex-shrink object-cover opacity-0"
                src="/src/assets/images/construction.png"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="rounded-xl bg-transparent  shadow-3d-light dark:shadow-3d-dark transition-all hover:shadow-hover-light dark:hover:shadow-hover-dark "
      style="grid-area: 21 / 14 / span 12 / span 36;"
    >
      <div
        class="flex w-full h-full"
        style="perspective: 1000px;"
      >
        <div
          class="w-full relative transition-all duration-200 ease-linear inter-var"
          style="transform-style: preserve-3d;"
        >
          <div
            class="[transform-style:preserve-3d] [&>*]:[transform-style:preserve-3d] bg-gray-50 group/card dark:hover:shadow-2xl dark:bg-black dark:border-white/[0.2] border-black/[0.1] w-full h-full rounded-xl border"
          >
            <div
              class="text-blue-900 rounded-xl flex justify-center flex-wrap items-center contain-size overflow-hidden h-full border-[2px] bg-app-lightModeBoxes-box10 dark:bg-app-darkModeBoxes-box10 border-app-darkModeBoxes-box10 dark:border-app-lightModeBoxes-box10"
              data-testid="example-container"
            >
              <div
                class="absolute inset-0 flex items-center justify-center"
              >
                <span
                  class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-1yonwsg-MuiCircularProgress-root"
                  role="progressbar"
                  style="width: 40px; height: 40px;"
                >
                  <svg
                    class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
                    viewBox="22 22 44 44"
                  >
                    <circle
                      class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-1a8epmm-MuiCircularProgress-circle"
                      cx="44"
                      cy="44"
                      fill="none"
                      r="20.2"
                      stroke-width="3.6"
                    />
                  </svg>
                </span>
              </div>
              <img
                class="rounded-lg min-h-full min-w-full flex-shrink object-cover opacity-0"
                src="/src/assets/images/construction.png"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="rounded-xl bg-transparent  shadow-3d-light dark:shadow-3d-dark transition-all hover:shadow-hover-light dark:hover:shadow-hover-dark "
      style="grid-area: 33 / 22 / span 22 / span 22;"
    >
      <div
        class="flex w-full h-full"
        style="perspective: 1000px;"
      >
        <div
          class="w-full relative transition-all duration-200 ease-linear inter-var"
          style="transform-style: preserve-3d;"
        >
          <div
            class="[transform-style:preserve-3d] [&>*]:[transform-style:preserve-3d] bg-gray-50 group/card dark:hover:shadow-2xl dark:bg-black dark:border-white/[0.2] border-black/[0.1] w-full h-full rounded-xl border"
          >
            <div
              class="flex flex-col justify-between h-full p-4  rounded-xl 
    bg-app-lightModeBoxes-box16 
    dark:bg-app-darkModeBoxes-box16 
    dark:border-app-lightModeBoxes-box16 
    border-app-darkModeBoxes-box16
    border-[2px]"
            >
              <div
                class="mb-4"
              >
                <h2
                  class="text-xl font-bold mb-2 text-black dark:text-white"
                >
                  My Favorite Playlist
                </h2>
                <p
                  class="text-black dark:text-white"
                >
                  Enjoy my personal selection of songs in this playlist. Listen to it and let yourself be carried away by the rhythm!
                </p>
              </div>
              <div
                class="mt-auto"
              >
                <iframe
                  allow="autoplay; clipboard-write; encrypted-media; fullscreen; picture-in-picture"
                  class="rounded-xl"
                  data-testid="spotify-playlist-iframe"
                  height="379px"
                  loading="lazy"
                  src="https://open.spotify.com/embed/playlist/2Z8vIZcBXeeCfcmkffrTlR?utm_source=generator"
                  width="100%"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="rounded-xl bg-transparent  shadow-3d-light dark:shadow-3d-dark transition-all hover:shadow-hover-light dark:hover:shadow-hover-dark "
      style="grid-area: 48 / 14 / span 6 / span 8;"
    >
      <div
        class="flex w-full h-full"
        style="perspective: 1000px;"
      >
        <div
          class="w-full relative transition-all duration-200 ease-linear inter-var"
          style="transform-style: preserve-3d;"
        >
          <div
            class="[transform-style:preserve-3d] [&>*]:[transform-style:preserve-3d] bg-gray-50 group/card dark:hover:shadow-2xl dark:bg-black dark:border-white/[0.2] border-black/[0.1] w-full h-full rounded-xl border"
          >
            <div
              class="text-blue-900 rounded-xl flex justify-center flex-wrap items-center contain-size overflow-hidden h-full border-[2px] bg-app-lightModeBoxes-box7 dark:bg-app-darkModeBoxes-box7 border-app-darkModeBoxes-box7 dark:border-app-lightModeBoxes-box7"
              data-testid="example-container"
            >
              <div
                class="absolute inset-0 flex items-center justify-center"
              >
                <span
                  class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-1yonwsg-MuiCircularProgress-root"
                  role="progressbar"
                  style="width: 40px; height: 40px;"
                >
                  <svg
                    class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
                    viewBox="22 22 44 44"
                  >
                    <circle
                      class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-1a8epmm-MuiCircularProgress-circle"
                      cx="44"
                      cy="44"
                      fill="none"
                      r="20.2"
                      stroke-width="3.6"
                    />
                  </svg>
                </span>
              </div>
              <img
                class="rounded-lg min-h-full min-w-full flex-shrink object-cover opacity-0"
                src="/src/assets/images/construction.png"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="rounded-xl bg-transparent  shadow-3d-light dark:shadow-3d-dark transition-all hover:shadow-hover-light dark:hover:shadow-hover-dark "
      style="grid-area: 33 / 44 / span 7 / span 20;"
    >
      <div
        class="flex w-full h-full"
        style="perspective: 1000px;"
      >
        <div
          class="w-full relative transition-all duration-200 ease-linear inter-var"
          style="transform-style: preserve-3d;"
        >
          <div
            class="[transform-style:preserve-3d] [&>*]:[transform-style:preserve-3d] bg-gray-50 group/card dark:hover:shadow-2xl dark:bg-black dark:border-white/[0.2] border-black/[0.1] w-full h-full rounded-xl border"
          >
            <div
              class="overflow-hidden relative"
              data-testid="frame-container"
            >
              <div
                class="transition-transform duration-500"
                style="transform: translateY(-0%); height: 152px;"
              >
                <div
                  style="pointer-events: none;"
                >
                  <iframe
                    allow="autoplay; clipboard-write; encrypted-media *; fullscreen; picture-in-picture"
                    class="rounded-xl"
                    data-testid="song-test-id-0"
                    height="152"
                    loading="lazy"
                    sandbox="allow-forms allow-popups allow-same-origin allow-scripts allow-storage-access-by-user-activation allow-top-navigation-by-user-activation"
                    src="https://open.spotify.com/embed/album/6H73t0Wok8dsEEGqRhgOsV?utm_source=generator"
                    style="pointer-events: auto; overflow: hidden; background: transparent;"
                    width="365"
                  />
                </div>
                <div
                  style="pointer-events: none;"
                >
                  <iframe
                    allow="autoplay; clipboard-write; encrypted-media *; fullscreen; picture-in-picture"
                    class="rounded-xl"
                    data-testid="song-test-id-1"
                    height="152"
                    loading="lazy"
                    sandbox="allow-forms allow-popups allow-same-origin allow-scripts allow-storage-access-by-user-activation allow-top-navigation-by-user-activation"
                    src="https://embed.music.apple.com/do/album/un-a%C3%B1o-single/1760344218"
                    style="pointer-events: auto; overflow: hidden; background: transparent;"
                    width="365"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="rounded-xl bg-transparent  shadow-3d-light dark:shadow-3d-dark transition-all hover:shadow-hover-light dark:hover:shadow-hover-dark "
      style="grid-area: 40 / 44 / span 10 / span 10;"
    >
      <div
        class="flex w-full h-full"
        style="perspective: 1000px;"
      >
        <div
          class="w-full relative transition-all duration-200 ease-linear inter-var"
          style="transform-style: preserve-3d;"
        >
          <div
            class="[transform-style:preserve-3d] [&>*]:[transform-style:preserve-3d] bg-gray-50 group/card dark:hover:shadow-2xl dark:bg-black dark:border-white/[0.2] border-black/[0.1] w-full h-full rounded-xl border"
          >
            <div
              class="text-blue-900 rounded-xl flex justify-center flex-wrap items-center contain-size overflow-hidden h-full border-[2px] bg-app-lightModeBoxes-box11 dark:bg-app-darkModeBoxes-box11 border-app-darkModeBoxes-box11 dark:border-app-lightModeBoxes-box11"
              data-testid="example-container"
            >
              <div
                class="absolute inset-0 flex items-center justify-center"
              >
                <span
                  class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-1yonwsg-MuiCircularProgress-root"
                  role="progressbar"
                  style="width: 40px; height: 40px;"
                >
                  <svg
                    class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
                    viewBox="22 22 44 44"
                  >
                    <circle
                      class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-1a8epmm-MuiCircularProgress-circle"
                      cx="44"
                      cy="44"
                      fill="none"
                      r="20.2"
                      stroke-width="3.6"
                    />
                  </svg>
                </span>
              </div>
              <img
                class="rounded-lg min-h-full min-w-full flex-shrink object-cover opacity-0"
                src="/src/assets/images/construction.png"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="rounded-xl bg-transparent  shadow-3d-light dark:shadow-3d-dark transition-all hover:shadow-hover-light dark:hover:shadow-hover-dark "
      style="grid-area: 41 / 54 / span 14 / span 10;"
    >
      <div
        class="flex w-full h-full"
        style="perspective: 1000px;"
      >
        <div
          class="w-full relative transition-all duration-200 ease-linear inter-var"
          style="transform-style: preserve-3d;"
        >
          <div
            class="[transform-style:preserve-3d] [&>*]:[transform-style:preserve-3d] bg-gray-50 group/card dark:hover:shadow-2xl dark:bg-black dark:border-white/[0.2] border-black/[0.1] w-full h-full rounded-xl border"
          >
            <div
              class="relative flex flex-col justify-between rounded-xl h-full"
            >
              <div
                class="absolute inset-0 flex justify-center items-center bg-gray-700 rounded-xl z-10"
              >
                <span
                  class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary text-white css-1yonwsg-MuiCircularProgress-root"
                  role="progressbar"
                  style="width: 40px; height: 40px;"
                >
                  <svg
                    class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
                    viewBox="22 22 44 44"
                  >
                    <circle
                      class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-1a8epmm-MuiCircularProgress-circle"
                      cx="44"
                      cy="44"
                      fill="none"
                      r="20.2"
                      stroke-width="3.6"
                    />
                  </svg>
                </span>
              </div>
              <video
                autoplay=""
                class="rounded-xl cursor-pointer opacity-0"
                loop=""
                playsinline=""
                title="Click to mute/unmute"
                width="210px"
              >
                <source
                  src="/src/assets/videos/guitar-video.mp4"
                  type="video/mp4"
                />
              </video>
              <div
                class="absolute bottom-3 left-2"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium text-white text-2xl css-20bmp1-MuiSvgIcon-root"
                  data-testid="VolumeOffIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M16.5 12c0-1.77-1.02-3.29-2.5-4.03v2.21l2.45 2.45c.03-.2.05-.41.05-.63m2.5 0c0 .94-.2 1.82-.54 2.64l1.51 1.51C20.63 14.91 21 13.5 21 12c0-4.28-2.99-7.86-7-8.77v2.06c2.89.86 5 3.54 5 6.71M4.27 3 3 4.27 7.73 9H3v6h4l5 5v-6.73l4.25 4.25c-.67.52-1.42.93-2.25 1.18v2.06c1.38-.31 2.63-.95 3.69-1.81L19.73 21 21 19.73l-9-9zM12 4 9.91 6.09 12 8.18z"
                  />
                </svg>
              </div>
              <div
                class="absolute -bottom-7 left-0 right-0 -z-10 transition-transform duration-1000 ease-in-out transform -translate-y-10 opacity-0"
              >
                <span
                  aria-label="See full video"
                  class="cursor-pointer w-full"
                  data-mui-internal-clone-element="true"
                >
                  <a
                    class="bg-gray-700 text-white w-full pt-4 pb-1 text-sm font-bold rounded-b-lg opacity-75 hover:opacity-80 transition-opacity block text-center"
                    href="https://www.youtube.com/watch?v=Mfxv1jmkWlk"
                    target="_blank"
                  >
                    Click me!
                  </a>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
